content/blog/2022/07/verify-signatures-with-gha-and-prefix.md

```diff
@@ -0,0 +1,96 @@
+(be it an individual or an organization) and the GitHub repository inside of which the GitHub Action has been
+executed.
+
+Starting from today, Kubewarden provides a convenient way to check signatures produced by GitHub actions.
+
+For this example, a Kubernetes cluster with Kubewarden already installed is required. The installation process is
+described in the [quick start guide](https://docs.kubewarden.io/quick-start).
+
+### GitHubAction
+
+```yaml 
+apiVersion: policies.kubewarden.io/v1
+kind: ClusterAdmissionPolicy
+metadata:
+  name: verify-image-signatures-policy
+spec:
+  module: registry://ghcr.io/kubewarden/policies/verify-image-signatures:v0.1.4
+  rules:
+  - apiGroups: [""]
+    apiVersions: ["v1"]
+    resources: ["pods"]
+    operations:
+    - CREATE
+    - UPDATE
+  mutating: true
+  settings:
+    signatures:
+      - image: "ghcr.io/raulcabello/app-example:*" # match all tags 
+        github_actions:
+        - owner: "raulcabello"
+          repo: "app-example" #optional. if omited all apps from the owner will ve valid.
+```
+
+This policy verifies all containers with an image that is `ghcr.io/raulcabello/app-example`. It will accept
+containers that were signed inside a GitHub action for the owner `raulcabello` and the app `app-example`.
+
+### KeylessPrefix
+
+`KeylessPrefix` is similar to the existing keyless verification , the only difference is that it will validate the 
+subject based on a prefix instead of an exact match. 
+
+``` yaml
+apiVersion: policies.kubewarden.io/v1
+kind: ClusterAdmissionPolicy
+metadata:
+  name: verify-image-signatures-policy
+spec:
+  module: registry://ghcr.io/kubewarden/policies/verify-image-signatures:v0.1.4
+  rules:
+  - apiGroups: [""]
+    apiVersions: ["v1"]
+    resources: ["pods"]
+    operations:
+    - CREATE
+    - UPDATE
+  mutating: true
+  settings:
+    signatures:
+      - image: "ghcr.io/raulcabello/app-example:*" # match all tags 
+        keyless_prefix:
+          - issuer: "https://token.actions.githubusercontent.com"
+            subject: "https://github.com/raulcabello/app-example/.github/workflows/ci.yml@refs/tags/" # match all tags
+``` 
+
+This will accept containers whose signature contains the issuer `https://token.actions.githubusercontent.com` and the 
+subject starts with `https://github.com/raulcabello/app-example/.github/workflows/ci.yml@refs/tags/`. 
+
+Like in the previous example this will accept the image `ghcr.io/raulcabello/app-example`, however we **don't** recommend using 
+`KeylessPrefix` for GitHub actions validation. The reason is that `KeylessPrefix` will validate using just the subject, 
+on the other hand `GithubActions` validates using the `github_workflow_repository` extension in the signature certificate. 
+We found out when signing our policies in a [shared workflow](https://github.com/kubewarden/github-actions/blob/v1/policy-release/action.yaml#L43) 
```

Comment:
I would reword this and remove the "we found out" tbf. The thing is, the OIDC token is working as intended by design, but one should use the verification correctly. I would word is similar to:

"When GitHub creates the OIDC token used for the signatures, it sets `subject` as the URL containing the GHA workflow code, which doesn't necessarily match **where** it has run in the case of reusable workflows. This is by design. If one wants to check for the repo where the job was run, corresponding to that workflow code, GitHub has added a x502 certificate extension `github_workflow_repository` that contains it."

(also, notice is reusable workflow, not shared workflow).

I would also mention that in the future we intend to fail validation if KeylessPrefix has issuers that match `"https://token.actions.githubusercontent.com"`.

Reply:
done https://github.com/kubewarden/kubewarden.io/pull/116/commits/a2d9cd50f55120dd0526944f402ed4eb5213da5a